src/pool/cli.c

```diff
@@ -770,6 +770,7 @@ pool_disconnect_cp(tse_task_t *task, void *data)
 		 */
 		D_ERROR("failed to notify agent of pool disconnect: "DF_RC"\n",
 			DP_RC(rc));
+		rc = 0;
```

Comment:
it doesn't make sense to me to log an error that is not an error on the operation. so maybe change to warn?
maybe another question here, are all errors non-fatal or some? should we fail on some errors, but ignore others? not sure if it's a good idea to ignore all errors..

Reply:
I could change to warn, although I do think it's useful to log at error, since it is a sign something is wrong with the daos_agent or with the client machine more generally (e.g. out of memory).

What happened here is the pool disconnect in fact succeeded. After we successfully disconnect a pool handle, we also send a notification to the daos_agent to update its local process monitor. That is the operation that fails here. At this point, the pool handle is gone on the server side, too, so the operation succeeded regardless of why the agent notify operation failed. Retrying the daos_pool_disconnect call will report -DER_NO_HDL.

For that reason I think it's okay to log the errors here and then ignore them. The side effect, as noted in the code comment, is that the agent will contact the server when the client process terminates, asking the server to clean up the already-gone handles that the agent thinks are still around. For the server this is a little inefficiency, as there will be nothing left to clean up. I guess in theory this could impact performance, which may be a good reason to log at error level for the client.